### PR TITLE
WIP Add custom time zones to delivered report

### PR DESCRIPTION
**Pull request by ALawley**
Here's what I have for #20 so far. I kept PDT set as a default time zone if there's none specified in the settings. Is that alright? 
I'm not certain what I want to be asserting in the test. Is there any way to view a sample delivered report pdf to see what its contents look like (since it looks like we're asserting that it contains a specified string)?

### CODE REVIEW

**Comment by kelseyq** [comment]
Pacific as default is great! There are two approaches you could take here to look at the generated result. One would be to temporarily add a line in report_delivery.send_report_to_school that writes the generated PDF to your filesystem. Another would be to [extract the text](https://github.com/SexualHealthInnovations/callisto-core/blob/master/tests/callistocore/test_delivery.py#L84) of the PDF in this test and print it to standard output

**Comment by ALawley** [comment]
Okay, I think I've got the test completed. `override_settings` doesn't seem to play nice with TIME_ZONE (https://github.com/pytest-dev/pytest-django/issues/93), so I had to use `timezone.activate()` and `timezone.deactivate()` for the test instead. Test passes currently, and fails when I change `tzname` on line [63](https://github.com/ALawley/callisto-core/blob/report-timezone/tests/callistocore/test_delivery.py#L63) to a different time zone.

**Comment by ALawley** [comment]
Wait, nevermind. It's passing when I change `tzname` in report_delivery.py back to the old hardcoded PDT version. Trying to figure out why that is.

**Comment by ALawley** [comment]
Okay, sorry about all the confusion with this. I've done some looking and I think I've got a better idea of what's going on now. `settings.TIME_ZONE`  pulls the default time zone, while `timezone.activate(timezone)` pulls the current timezone (which uses the default time zone as a default if no current one is set). 

For whatever reason, setting the current time zone in the test seems to override any time zones in the method we're testing (I'm not sure why this is). So the current time isn't usable in the test. But the default time isn't usable either because we can't set it (unless I'm misunderstanding the issue I linked, which is very possible, but then I'm still having trouble getting the `override_settings` decorator to work).

So I'm not exactly sure what to do. The best thing I can think of is that since the default default time zone is `America/Chicago` I could not bother setting the time zone in the test and have the test expect Chicago? But that doesn't seem like a very good test. Do you have any ideas? I feel like there has to be something I'm missing here.

**Comment by kelseyq** [comment]
I should have known better than to think anything involving timezones would have a straightforward path!!!!

So, stepping back a bit: the thing we care about functionally here is that it's possible to set a timezone once and have the report use that timezone. Since Django already has Central time as a default, I propose that we set `tzname` in `report_delivery.py` to `settings.TIME_ZONE` with no default. In the tests' [settings file](https://github.com/SexualHealthInnovations/callisto-core/blob/master/tests/settings.py), we specify a non-Chicago timezone and we include a test that verifies that that timezone is used in the generated report. What do you think? Are there any use cases or failure scenarios that we miss that way?

**Comment by ALawley** [comment]
Oh, I didn't know about the tests settings file. That should fix the issues I was having with setting the time zone at run time in the test! I think that that test should cover what we need. I'll go ahead and write it a bit later this afternoon.

**Comment by ALawley** [comment]
Alright, the new test should be what we want now. Thanks for the pointer to the test settings! If it looks good to you, that's both of the low-hanging fruit issues done. Do you have a suggestion for a next issue for me to work on?

**Comment by kelseyq** [comment]
You're a MACHINE! Want to take a crack at #72? Let me know if I can clarify it further.

**Comment by kelseyq** [comment]
Hey Anna--I looked into the Django `settings.TIME_ZONE` setting and it's not quite what we want--it specifies a timezone for datetimes [stored in the DB](https://docs.djangoproject.com/en/1.9/ref/settings/), and we want to store times in the DB as UTC (fun summary of why [here](http://stackoverflow.com/a/2532962/1308611), if your values of fun include misery...) Can you drop in a different setting (`settings.REPORT_TIME_ZONE` or something) instead here?

**Comment by ALawley** [comment]
Okay, I've got it pulling from setting.REPORT_TIME_ZONE and adjusted the test settings to match. Although will REPORT_TIME_ZONE use Chicago as the default in the same way that TIME_ZONE does? Do I need to add an or statement back in to `tzname` so we have a default if REPORT_TIME_ZONE is undefined?

File: tests/callistocore/test_delivery.py [hunk 1]
```diff
@@ -48,6 +51,18 @@ def test_pdf_full_report_is_generated(self):
         self.assertIn("test answer", pdfReader.getPage(1).extractText())
         self.assertIn("answer to 2nd question", pdfReader.getPage(1).extractText())
 
+    def test_pdf_report_generated_with_timestamp(self):
+        # test_tzname matches TIME_ZONE in tests/settings.py
```
Comment (kelseyq) [inline tests/callistocore/test_delivery.py]: does `override_settings` still not work now that we're pulling a string from settings?

**Comment by kelseyq** [comment]
Yeah, can you add back the default to Pacific? It'll help with rolling this out to our instances that are already in production (which happen to both be in PST.) Thanks!

**Comment by ALawley** [comment]
`@override_settings(REPORT_TIME_ZONE='Europe/Paris')` still isn't working and I'm not sure why. When I change `REPORT_TIME_ZONE` in `settings.py` to `'America/Los_Angeles'` and override it in the test to Paris, the test returns a PST PDF. So I've just got the test running the way it was, just using `settings.py`. I've added PST back as the default for `tzname`.

**Comment by kelseyq** [comment]
Ah, okay. The variable might be getting set before the test settings are overridden--I'm still a bit shaky on Python instantiation stuff.

**Comment by kelseyq** [comment]
Looks good to me! I'm merging this and we'll be using this feature in the new school rollouts happening next month--thanks so much!

**Pull request merged**